src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -2749,6 +2878,160 @@ private void ensurePublishCapability(
         }
     }
 
+    private final class MqttConnectPayload
+    {
+        private int reasonCode;
+
+        private final MqttWillProperties willPropertiesRO = new MqttWillProperties();
+
+        private String16FW willTopic = null;
+        private OctetsFW willPayload = null;
+        private String16FW username = null;
+        private OctetsFW password = null;
+
+        private void reset()
+        {
+            this.reasonCode = NORMAL_DISCONNECT;
+            this.willPropertiesRO.reset();
+            this.willTopic = null;
+            this.willPayload = null;
+            this.username = null;
+            this.password = null;
+        }
+
+        private void decodePayload(
+            MqttConnectFW connect)
+        {
```

Comment:
Suggest combining these 2 methods into `decode(MqttConnectFW)`, and adding a return type of `MqttConnectPayload`.

Then the caller will do something like:
```
MqttConnectPayload payload = mqttConnectPayloadRO.decode(connect);
payload.reasonCode ...
```